### ISSUE TITLE
web3-provider has MaxListenersExceededWarning

### ISSUE THREAD

**Issue by 0xSSoul** [WalletConnect/walletconnect-monorepo#295]
web3-provider version 1.0.3

Several minutes after connecting, I gets warning message in Chrome console:
```
events.js?faa1:46 MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 31 error listeners added. Use emitter.setMaxListeners() to increase limit
    at _addListener (webpack-internal:///./node_modules/events/events.js:210:15)
    at WalletConnectProvider.addListener (webpack-internal:///./node_modules/events/events.js:226:10)
    at eval (webpack-internal:///./node_modules/@walletconnect/web3-provider/dist/cjs/index.js:299:22)
    at new Promise (<anonymous>)
    at WalletConnectProvider.eval (webpack-internal:///./node_modules/@walletconnect/web3-provider/dist/cjs/index.js:293:20)
    at Generator.next (<anonymous>)
    at eval (webpack-internal:///./node_modules/tslib/tslib.es6.js:98:71)
    at new Promise (<anonymous>)
    at Module.__awaiter (webpack-internal:///./node_modules/tslib/tslib.es6.js:94:12)
    at WalletConnectProvider.handleReadRequests (webpack-internal:///./node_modules/@walletconnect/web3-provider/dist/cjs/index.js:286:24)
```

I think it is because the handleReadRequests() add the listeners by this.on() again and again, but never off() those bindings.

```
 async handleReadRequests(payload: any): Promise<IJsonRpcResponseSuccess> {
    if (!this.http) {
      const error = new Error("HTTP Connection not available");
      this.emit("error", error);
      throw error;
    }
    this.http.send(payload);
    return new Promise((resolve, reject) => {
      this.on("payload", (response: IJsonRpcResponseSuccess) => {
        if (response.id === payload.id) {
          resolve(response);
        }
      });
      this.on("error", (response: IJsonRpcResponseError) => {
        if (response.id === payload.id) {
          reject(response.error.message);
        }
      });
    });
  }
```

**Comment by pedrouid** [comment]
Good point, I've published a test version on v1.0.6. Could you give it a try on your side?

**Comment by 0xSSoul** [comment]
I tried v1.0.7 and reviewed the code. I think the problem is fixed.
